features/step_definitions/admin_node/packages_steps.rb

```diff
@@ -9,6 +9,13 @@
 end
 
 Given(/^all dependencies of installed packages are satisfied$/) do
-  admin_node.exec!("zypper", "verify --no-recommends")
+  wait_for "zypper verify --no-recommends", max: "2 minutes", sleep: "5 seconds" do
+    result = admin_node.exec!("zypper", "verify --no-recommends", capture_error: true)
+    if result.exit_code == 7 # ZYPPER_EXIT_ZYPP_LOCKED
+      log.warn(result.error)
+      next
+    end
+    break if result.exit_code.zero?
```

Comment:
Breaking here means escaping the timeout loop, but you are right that there is an exception missing for other codes.  
I will however use the `export ZYPP_LOCK_TIMEOUT=120` and replace this `wait_for` completely.